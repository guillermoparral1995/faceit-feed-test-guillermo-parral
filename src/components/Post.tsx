import Image from "next/image";
import Link from "next/link";

interface PostProps {
  post: Post;
  withHighlight: boolean;
}

export default function Post({ post, withHighlight }: PostProps) {
  return (
    <article
      className={`cursor-pointer max-w-3xl mx-auto dark:prose-dark rounded-md border-2 p-4 my-4 ${
        withHighlight ? "bg-yellow-500" : ""
      }`}
    >
      <Link className="flex gap-4" href={`/${post.id}`}>
        <div className="flex-none">
          <Image
            src={post.author.avatar.thumbnail}
            alt={post.author.name}
            width={40}
            height={40}
          />
        </div>
        <div className="flex-auto">
          <h3 className="mb-2 font-bold">{post.author.name}</h3>
          <p className="line-clamp-2 font-normal">{post.post.body}</p>
        </div>
      </Link>
    </article>
  );
}
