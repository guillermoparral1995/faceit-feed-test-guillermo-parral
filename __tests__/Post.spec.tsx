import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Post from "@/components/Post";

const mockPost: Post = {
  id: 1,
  author: {
    name: "John Doe",
    avatar: {
      thumbnail: "https://example.com/avatar.jpg",
      picture: "https://example.com/avatar.jpg",
    },
  },
  post: {
    created_at: new Date("2024-03-03").toISOString(),
    body: "Hello, world!",
  },
};

describe("Post", () => {
  it("should render correctly - normal", () => {
    const component = render(<Post post={mockPost} withHighlight={false} />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly - with highlight", () => {
    const component = render(<Post post={mockPost} withHighlight />);

    expect(component).toMatchSnapshot();
  });
});
