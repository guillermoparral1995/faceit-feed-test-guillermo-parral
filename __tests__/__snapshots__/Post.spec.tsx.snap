// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly - normal 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="cursor-pointer max-w-3xl mx-auto dark:prose-dark rounded-md border-2 p-4 my-4 transition duration-500 bg-transparent"
      >
        <a
          class="flex gap-4"
          href="/1"
        >
          <div
            class="flex-none"
          >
            <img
              alt="John Doe"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
          <div
            class="flex-auto"
          >
            <h3
              class="mb-2 font-bold"
            >
              John Doe
            </h3>
            <p
              class="line-clamp-2 font-normal"
            >
              Hello, world!
            </p>
          </div>
        </a>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="cursor-pointer max-w-3xl mx-auto dark:prose-dark rounded-md border-2 p-4 my-4 transition duration-500 bg-transparent"
    >
      <a
        class="flex gap-4"
        href="/1"
      >
        <div
          class="flex-none"
        >
          <img
            alt="John Doe"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="flex-auto"
        >
          <h3
            class="mb-2 font-bold"
          >
            John Doe
          </h3>
          <p
            class="line-clamp-2 font-normal"
          >
            Hello, world!
          </p>
        </div>
      </a>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Post should render correctly - with highlight 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="cursor-pointer max-w-3xl mx-auto dark:prose-dark rounded-md border-2 p-4 my-4 transition duration-500 bg-yellow-500"
      >
        <a
          class="flex gap-4"
          href="/1"
        >
          <div
            class="flex-none"
          >
            <img
              alt="John Doe"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
          <div
            class="flex-auto"
          >
            <h3
              class="mb-2 font-bold"
            >
              John Doe
            </h3>
            <p
              class="line-clamp-2 font-normal"
            >
              Hello, world!
            </p>
          </div>
        </a>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="cursor-pointer max-w-3xl mx-auto dark:prose-dark rounded-md border-2 p-4 my-4 transition duration-500 bg-yellow-500"
    >
      <a
        class="flex gap-4"
        href="/1"
      >
        <div
          class="flex-none"
        >
          <img
            alt="John Doe"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="flex-auto"
        >
          <h3
            class="mb-2 font-bold"
          >
            John Doe
          </h3>
          <p
            class="line-clamp-2 font-normal"
          >
            Hello, world!
          </p>
        </div>
      </a>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
